lldb/tools/driver/Driver.cpp

```diff
@@ -423,6 +426,47 @@ SBError Driver::ProcessArgs(const opt::InputArgList &args, bool &exiting) {
   return error;
 }
 
+#ifdef _WIN32
+// Resolve the Embeddable Python directory bundled with the Swift toolchain.
+// If it exists, add it to the list of DLL search directories.
+// The installation of Embeddable Python is optional when installing the Swift
+// toolchain, therefore the directory might not exist.
+void AddPythonDLLToSearchPath() {
+  wchar_t buffer[MAX_PATH];
+  DWORD length = GetModuleFileNameW(NULL, buffer, MAX_PATH);
```

Comment:
This is risky - the path is possibly a long path (i.e. >MAX_PATH). In such a case, LLDB will be unusable permanently. The path returned is the path used to load, so this is not something we can control.

It is likely best to use `GetModuleHandleW` and then `GetFinalPathNameByHandleW` to get the path.

Reply:
I was not able to use `GetModuleHandleW` with `GetFinalPathNameByHandleW` directly after, there had to be an intermiedary `GetModuleFileNameW` and it was back to the original problem.

I found [this SO thread](https://stackoverflow.com/questions/805814/how-can-i-calculate-the-complete-buffer-size-for-getmodulefilename) where others are struggling with this API. I settled with the _grow the buffer until it fits_ solution.